lib/polly.ml

```diff
@@ -70,11 +75,17 @@ let create = caml_polly_create1
 
 let close t = Unix.close t
 
-let add = caml_polly_add
+let add t fd e =
+  let r = caml_polly_add t fd e in
+  if r = -1 then caml_raise_unix_error "Polly.caml_polly_add"
```

Comment:
It might be a bit more defensive to test for r < 0.

Reply:
OK